components/dashboard/src/admin/TeamDetail.tsx

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import moment from "moment";
+import { useEffect, useState } from "react";
+import { Team, TeamMemberInfo, TeamMemberRole } from "@gitpod/gitpod-protocol";
+import { getGitpodService } from "../service/service";
+import { Item, ItemField, ItemsList } from "../components/ItemsList";
+import PillLabel from "../components/PillLabel";
+import DropDown from "../components/DropDown";
+import { Link } from "react-router-dom";
+
+export default function TeamDetail(props: { team: Team }) {
+    const { team } = props;
+    const [teamMembers, setTeamMembers] = useState<TeamMemberInfo[] | undefined>(undefined);
+    const [searchText, setSearchText] = useState<string>('');
+
+    useEffect(() => {
+        (async () => {
+            const members = await getGitpodService().server.adminGetTeamMembers(team.id);
+            if (members.length > 0) {
+                setTeamMembers(members)
+            }
+        })();
+    }, [team]);
+
+    const filteredMembers = teamMembers?.filter(m => {
+        const memberSearchText = `${m.fullName || ''}${m.primaryEmail || ''}`.toLocaleLowerCase();
+        if (!memberSearchText.includes(searchText.toLocaleLowerCase())) {
+            return false;
+        }
+        return true;
+    });
+
+    const setTeamMemberRole = async (userId: string, role: TeamMemberRole) => {
+        await getGitpodService().server.adminSetTeamMemberRole(team!.id, userId, role);
+        setTeamMembers(await getGitpodService().server.adminGetTeamMembers(team!.id));
+    }
+    return <>
+        <div className="flex">
+            <div className="flex-1">
+                <div className="flex"><h3>{team.name}</h3>
+                    {team.markedDeleted && <span className="mt-2"><PillLabel type="warn" className="font-semibold mt-2 py-0.5 px-2 self-center">Deleted</PillLabel></span>}
```

Comment:
**issue(non-blocking)**: While using a label like this here sounds great, we could use the same design for the blocked user labed as described in https://github.com/gitpod-io/gitpod/issues/8060#issuecomment-1047869502. However, it's also fine to improve in one area and let the previous design catch-up.

**question**: Wondering how long it takes for soft-deleted items to disappear.

Reply:
1. Will change it to the same design as the Blocked label.
2. Teams, and also other entities in our project, are not technically "soft"-deleted but only "marked" as deleted. The rows will stay in the database, but they will be excluded from being shown. So the answer is, they won't disappear.